llm-chain/src/text_splitter.rs

```diff
@@ -0,0 +1,138 @@
+use std::cmp::max;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TextSplitError {
+    #[error("Error tokenizing input text")]
+    TokenizationError,
+    #[error("Error converting token chunk to string")]
+    ChunkToStringError,
+}
+
+pub trait TextSplitter<TokenType: Clone> {
```

Comment:
Same things with 
https://github.com/sobelio/llm-chain/blob/09f698af41be963d1c3d4f481a1ef656c5f1ac9d/llm-chain/src/tokens.rs#L37-L83

Can we make this use you system